test/libsolidity/syntaxTests/nameAndTypeResolution/584_abi_decode_with_tuple_of_other_than_types.sol

```diff
@@ -2,4 +2,4 @@ contract C {
     function f() pure public { abi.decode("", (0)); }
 }
 // ----
-// TypeError 1039: (60-61): Argument has to be a type name.
+// TypeError 5901: (60-61): Expression is not allowed as argument here. Argument must be a built-in type name, an user-defined type name or an array of those.
```

Comment:
Is there even a case where can still hit error 1039? If not, I'd replace it with an assert and reuse its number. And even if we can, i don't see much point in distinguishing them. We could have a single message covering both.

And the message should be adjusted since technically a type name is still an expression. I'd keep just the second sentence.

Reply:
The new approach (`mobileType` overloaded for `TypeType`) removes error `5901` and its message altogether.